pkg/sql/codegen/intermediate_representation.go

```diff
@@ -59,12 +59,25 @@ type FeatureColumn interface {
 	GetFieldMeta() []*FieldMeta
 }
 
+// SQLProgramIR represents a parsed SQL program.
+// TODO(typhoonzero): Can generate a DAG workflow from a SQL program.
+type SQLProgramIR []SingleSQLIR
+
+// SingleSQLIR represent all kind of IRs including: TrainIR, PredictIR, AnalyzeIR and standard SQL.
+type SingleSQLIR interface {
+	// This function is used only for restrict the IR struct types
+	IsIR()
+}
+
 // TrainIR is the intermediate representation for code generation of a training job.
 //
 // Please be aware that the TrainIR intentionally excludes the model table name in the
 // INTO clause. The sql package will save the output files of a generated Python program.
 // For prediction and analysis jobs, the sql will restore an identical working directly.
 type TrainIR struct {
+	// OriginalSQL record the original SQL statement used to get current IR result
+	// FIXME(typhoonzero): OriginalSQL is a temporary field. Can remove this when all moved to IR
```

Comment:
What is "all" in "all moved to IR"?  Does it mean `ALPS` and `EDL`?

`executor_ir.go: 146` also uses `OriginalSQL` during the model saving.